homework/src/app.js

```diff
@@ -1,11 +1,254 @@
 {
   /* Your homework goes here */
-  
   /* STRINGS */
-
   /* ARRAYS */
-
   /* OPERATORS */
-
   /* QUESTIONS FROM HYF REPOS */
-}
\ No newline at end of file
+}
+
+// STRINGS
+// Strings Question 1
+
+//  Consider the following string:
+
+const myString = 'hello,this,is,a,difficult,to,read,sentence';
+
+// Q: Add the string to your file and log it.
+
+// Solution
+
+console.log(myString);
+
+// Log the length of myString.
+
+// Solution : we use length property for the solution.
+
+console.log(myString.length);
+
+// Q:The commas make that the sentence is quite hard to read. Find a way to remove the commas from the string and replace them with spaces.
+
+// Solution
+
+const myString = 'hello,this,is,a,difficult,to,read,sentence';
+
+let regex = /,/gi;
+
+console.log(myString.replace(regex, ' '));
+
+// Strings Question 2:
+
+// Consider the following string:
+
+const str = 'dlroW olleH';
+
+// Q:The string doesn't make any sense. Find a way to reverse this string so that it becomes readable.
+
+// first i want to turn the string to an array, for that purpose I used split property
+let myNewArray = str.split('');
+
+console.log(myNewArray);
+
+// then i want to use reverse property to make the it readable and meaningful
+let mySecondArray = myNewArray.reverse();
+
+console.log(mySecondArray);
+
+// Now I want to get rid off the commas;
+
+console.log(mySecondArray.replace(regex, ' '));
+
+// Arrays
+
+// 1. Consider the following array:
+
+const favoriteAnimals = ['blowfish', 'capricorn', 'giraffe'];
+
+// Add a statement that adds your favorite animal say 'turtle' to the existing array.
+
+favoriteAnimals.push('turtle');
+
+// Explain in a comment what do you think the new array will look like.
+
+// i think my new array will be ["blowfish", "capricorn", "giraffe", "turtle"]; because push property add the new element at the end of the current array.
+
+// Log your new array.
+
+console.log(favoriteAnimals);
+
+// Now add another favorite animal to the array but make sure it is placed after 'blowfish' and before 'capricorn'.
+favoriteAnimals.splice(1, 0, 'lion');
+
+// Explain in a comment what do you think the new array will look like.
+
+// i think the new array will be : ["blowfish", "lion" "capricorn", "giraffe", "turtle"]
+
+// Log your new array.
+console.log(favoriteAnimals);
+
+// Delete 'giraffe' from the array.
+
+favoriteAnimals.splice(3, 1);
+
+// OPERATORS
+
+// Consider the following snippet:
+
+0 == '0'; // true
+0 == []; // true
+'0' == []; // false
+
+// Mathematically, if a == b and b == c, then a == c. Can you explain in your words why this is not true for the above snippet?
+
+0 == '0'; // true (although their types are not the same their values are equal; thats why its true.)
+0 == []; // true (== coerce empty arrays into its numeric value, which its length, so the length of [] is 0; that's why it is true)
+'0' == []; // false (i guess the object [] is not equal to a string thats why its false.)
```

Comment:
Ahaan - actually when we compare '0' == [] or 0 == [] - JavaScript will try to convert the empty array to a string using toString i.e [].toString() which is equal to an empty string. Thus, 0 == [] is essentially comparing 0 == "" which is true since both sides are falsy. However, '0' == [] results in '0' == '' which is false since right side is falsy but left side is truthy. Does it make sense? 🙂

Try going through this documentation and see if it helps - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Comparison_Operators